src/Patch/Manager.php

```diff
@@ -18,158 +18,88 @@
 class Manager
 {
     /**
-     * Directory for hotfixes.
-     */
-    const HOTFIXES_DIR = 'm2-hotfixes';
-
-    /**
-     * @var Applier
+     * @var LoggerInterface
      */
-    private $applier;
+    private $logger;
 
     /**
-     * @var LoggerInterface
+     * @var ShellInterface
      */
-    private $logger;
+    private $shell;
 
     /**
      * @var File
      */
     private $file;
 
     /**
-     * @var FileList
+     * @var DirectoryList
      */
-    private $fileList;
+    private $directoryList;
 
     /**
-     * @var DirectoryList
+     * @var GlobalSection
      */
-    private $directoryList;
+    private $globalSection;
 
     /**
-     * @param Applier $applier
      * @param LoggerInterface $logger
+     * @param ShellInterface $shell
      * @param File $file
-     * @param FileList $fileList
      * @param DirectoryList $directoryList
+     * @param GlobalSection $globalSection
      */
     public function __construct(
-        Applier $applier,
         LoggerInterface $logger,
+        ShellInterface $shell,
         File $file,
-        FileList $fileList,
-        DirectoryList $directoryList
+        DirectoryList $directoryList,
+        GlobalSection $globalSection
     ) {
-        $this->applier = $applier;
+
         $this->logger = $logger;
+        $this->shell = $shell;
         $this->file = $file;
-        $this->fileList = $fileList;
         $this->directoryList = $directoryList;
+        $this->globalSection = $globalSection;
     }
 
     /**
      * Applies all needed patches.
      *
      * @throws ShellException
-     * @throws FileSystemException
-     */
-    public function applyAll()
-    {
-        $this->copyStaticFile();
-        $this->applyComposerPatches();
-        $this->applyHotFixes();
-    }
-
-    /**
-     * Copying static file endpoint.
-     * This resolves issue MAGECLOUD-314
-     *
-     * @return void
      */
-    private function copyStaticFile()
+    public function apply()
     {
         $magentoRoot = $this->directoryList->getMagentoRoot();
 
         if (!$this->file->isExists($magentoRoot . '/pub/static.php')) {
-            $this->logger->notice('File static.php was not found.');
-
-            return;
+            $this->logger->notice('File static.php was not found');
```

Comment:
I think we should move this functionality from here, this is not related to patch applying.

Reply:
I'll keep in this 2002.0 branch and refactor in PR to `develop` as it does not give any profit

Reply:
ok, agree